src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.cpp

```diff
@@ -211,7 +211,8 @@ void ov::npuw::IBaseInferRequest::infer() {
 
 void ov::npuw::IBaseInferRequest::dump_input_tensors(std::size_t idx) {
     const std::string dump_ios_opt = m_npuw_model->m_cfg.get<::intel_npu::NPUW_DUMP_IO>();
-    if (!ov::npuw::util::is_set(idx, dump_ios_opt)) {
+    const std::size_t last_idx = m_npuw_model->m_compiled_submodels.size() - 1;
```

Comment:
maybe it makes sense to make `last_idx` be `end_idx` - exclusive, like `end()` iterator - so that you have `-1` in just a single place.

Reply:
Do you mean to add method to CompiledModel class which return the index of the last submodel?

Reply:
Might be Dmitry means just to rename `last_idx` to `end_idx` and use it without `-1` when passing to the `ov::npuw::util::is_set(idx, dump_ios_opt, end_idx)` while substracting 1 right in `is_set` function?

Reply:
Fixed